### PR TITLE
Add details on referencing other collections, and additional detail on other types

### PR DESCRIPTION
**Pull request by calummoore**
@mateuszmlc  Could you double check this is all valid?

### CODE REVIEW

File: collections.mdx [hunk 1]
```diff
@@ -132,29 +195,192 @@ The following types are supported:
...
+
+`Collection` fields reference another collection in the same namespace. This allows you to create relationships between records in the same or different collections. 
+Similar to how you have relationships in SQL.
+
+You can define the field on the collection, and then also pass a parameter as a `Collection` type.
 
 ```js
-@public
-collection CollectionName {
+@public Pet {
```
Comment (mateuszmlc) [inline collections.mdx:250]: Missing `collection`

File: collections.mdx [hunk 2]
```diff
@@ -132,29 +195,192 @@ The following types are supported:
...
+
+<Note>
+  Public keys on Polybase use non-prefixed Public Keys, so they are 64 bytes long.
+</Note>
+
+<Info>We will be adding support for [Falcon](https://eprint.iacr.org/2022/1041.pdf), this is a zk optimized PublicKey</Info>
+
+#### Arrays
+
+You can create arrays of `string`, `number`, `boolean`, `PublicKey` and `Collection` type. For example:
```
Comment (mateuszmlc) [inline collections.mdx:322]:
```suggestion
You can create arrays of `string`, `number`, `boolean`, `bytes` and `Collection` type. For example:
```

`PublicKey[]` is not supported right now. `bytes[]` is allowed.

File: write-data.mdx [hunk 3]
```diff
@@ -18,7 +22,11 @@ const collectionReference = db.collection("City");
 
 async function createRecord () {
   // .create(args) args array is defined by the constructor fn
-  const recordData = await collectionReference.create(["new-york", "New York"]);
+  const recordData = await collectionReference.create([
+    "new-york", 
+    "New York", 
+    db.collection("Person").record("johnbmahone")
```
Comment (mateuszmlc) [inline write-data.mdx:28]: The returned JSON example below is missing this (value would be `{ "collectionId": "your-namespace/Person", "id": "johnbmahone" }`)

**Review by mateuszmlc — CHANGES_REQUESTED** [review]

File: collections.mdx (hunk 2 above)
Comment (calummoore) [reply collections.mdx:322]: Ah okay, will create a ticket. Workaround right now would be to delegate to User[] then.

**Pull request merged**